modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt

```diff
@@ -0,0 +1,33 @@
+package com.sphereon.oid.fed.common.logic
+
+import com.sphereon.oid.fed.openapi.models.EntityStatement
+
+class EntityLogic {
+
+    fun getEntityType(entityStatement: EntityStatement): EntityType {
+        if (isFederationListEndpointPresent(entityStatement) == true && isAuthorityHintPresent(entityStatement) == false) {
+            return EntityType.TRUST_ANCHOR
+        } else if (isFederationListEndpointPresent(entityStatement) == true && isAuthorityHintPresent(entityStatement) == true) {
+            return EntityType.INTERMEDIATE
+        } else if (isFederationListEndpointPresent(entityStatement) == false && isAuthorityHintPresent(entityStatement) == true) {
+            return EntityType.LEAF
+        } else {
+            return EntityType.UNDEFINED
+        }
+    }
+
+    private fun isAuthorityHintPresent(entityStatement: EntityStatement): Boolean {
+        return entityStatement.authorityHints?.isEmpty() == false
+    }
+
+    private fun isFederationListEndpointPresent(entityStatement: EntityStatement): Boolean {
+        return entityStatement.metadata?.federationEntity?.federationListEndpoint?.isNotEmpty() == true
+    }
+}
+
+enum class EntityType {
+    LEAF,
+    INTERMEDIATE,
+    TRUST_ANCHOR,
+    UNDEFINED
+}
```

Comment:
Add trailing new line

Reply:
Are we following any specific style? Isn't this style for Python?

Reply:
I have made the changes

Reply:
Yes, Google Style. AFAIK it's not Python specific